### ISSUE TITLE
Building with mingw against EPICS 7.0.4

### ISSUE THREAD

**Issue by mfurseman** [epics-modules/pcas#6]
I found that when cross compiling with Mingw for Windows that the build fails with the following error:

```
/usr/bin/i686-w64-mingw32-g++              -D_MINGW -D__USE_MINGW_ANSI_STDIO      -O3   -Wall       -m32       -I. -I../O.Common -I. -I. -I.. -I../../../../src/pcas/generic -I../../../../src/pcas/io/bsdSocket -I../../../../src/pcas/generic/st -I/mnt/build/prefix/include/compiler/gcc -I/mnt/build/prefix/include/os/WIN32 -I/mnt/build/prefix/include -I/mnt/build/prefix/include/compiler/gcc -I/mnt/build/prefix/include/os/WIN32 -I/mnt/build/prefix/include                                                  -o casDGIntfIO.o -c ../../../../src/pcas/io/bsdSocket/casDGIntfIO.cc
In file included from /mnt/build/prefix/include/caProto.h:21:0,
                 from ../../../../src/pcas/generic/caHdrLargeArray.h:26,
                 from ../../../../src/pcas/generic/casCtx.h:19,
                 from ../../../../src/pcas/generic/casCtx.cc:15:
/mnt/build/prefix/include/os/WIN32/osdSock.h:83:1: error: 'LIBCOM_API' does not name a type; did you mean 'LPBITMAP'?
 LIBCOM_API unsigned epicsStdCall wsaMajorVersion ();
 ^~~~~~~~~~
 ```

I'm building against EPICS 7.0.4, where the release notes suggest some changes might be needed to continue LIBCOM support.

> Internally, the Com and ca libraries now express dllimport/export (Windows) and symbol visibility (GCC) using library-specific macros (eg. LIBCOM_API) instead of the macros epicsShareFunc, epicsShareClass, epicsShareDef etc. that are defined in the shareLib.h header. This change may affect some user code which uses the epicsShare* macros without having explicitly included the shareLib.h header themselves. Such code should be changed to include shareLib.h directly.
>> https://epics-base.github.io/epics-base/releasenotes.html

I've resolved this with https://github.com/mfurseman/pcas/commit/3ce37566ddc899b9d67febd9d4d3e4323529251d, although I'm not at all sure this is the right way of doing it.

**Comment by ralphlange** [comment]
Well, the CI builds are against 7.0.6.1 and don't show this issue.
Could it be that the issue has been fixed in Base?

**Comment by ralphlange** [comment]
Well, commit https://github.com/epics-base/epics-base/commit/799e72b1e338ffb79e1c1f4699a3d56f0cd43706 which handles the necessary changes in Base should be part of 7.0.4.
Specifically, [this line](https://github.com/epics-base/epics-base/blob/91941af992f6c32ef4f4ef55a2952f581aeee36a/modules/libcom/src/osi/osiSock.h#L19) in `osiSock.h` (which is included through `caProto.h`) should do pretty much what your proposed fix does.

Are you sure that your installation is consistent? Maybe your build includes an older `osiSock.h`?

**Comment by mfurseman** [comment]
Thanks Ralph, you've helped me work out what my problem is!

I was not really building against `R7.0.4` but against whichever commit was at the `HEAD` when we started this project (https://github.com/epics-base/epics-base/commit/1cacd058cd3bd1d9b6cabc6219c48217913eb1e8). This is two commits before https://github.com/epics-base/epics-base/commit/4dad8ca50351cec1a6addcc8edb89f4ba46ee818!

I've now built successfully against `R7.0.5` and `R7.0.6` (`R7.0.6.1` has some `_mkgmtime32 `issue for me).
